pkg/collector/resourceutilization/bpf/process_bpf_collector.go

```diff
@@ -87,19 +90,28 @@ func UpdateProcessBPFMetrics(bpfExporter bpf.Exporter, processStats map[uint64]*
 		return
 	}
 	for _, ct := range processesData {
-		comm := C.GoString((*C.char)(unsafe.Pointer(&ct.Comm)))
+		var comm string
+		if _, err := os.Stat("/proc/" + strconv.Itoa(int(ct.Pid))); os.IsNotExist(err) {
+			// skip process that is not running
+			continue
+		}
+		if commBytes, err := os.ReadFile("/proc/" + strconv.Itoa(int(ct.Pid)) + "/comm"); err == nil {
+			comm = strings.TrimSpace(unix.ByteSliceToString(commBytes))
```

Comment:
isn't reading from `/proc` expensive ? instead, can't we pass this data from kernel itself as it used to be? 
Also ,should we re-read the proc every time we process data?

Reply:
Re: Getting comm from eBPF... See:
https://github.com/sustainable-computing-io/kepler/issues/1354
https://github.com/sustainable-computing-io/kepler/issues/1640

It's "wrong" for 2 reasons:

1. https://github.com/sustainable-computing-io/kepler/blob/19c6afdccaa602d0a619c251f7ea17d9adb2b2c5/bpf/kepler.bpf.h#L281 `current_comm` is the process that is going OFF CPU, not the process going ON CPU 🐛 
2. Even if we were to use this helper correctly, we probably don't want to be getting the comm on every task switch since it never changes.

Not only that, but `comm` from eBPF is limited to 16 bytes (15 bytes + NUL).
The output from procfs is much nicer, see below.

```
kepler_process_cpu_instructions_total{command="postgres",container_id="fbe5d970ec22cf266028f676f98b5670e1193e33d571eb54f4c9f99f991a2d72",pid="1210084",source="bpf",vm_id=""} 971473
kepler_process_cpu_instructions_total{command="postgres: autovacuum launcher",container_id="fbe5d970ec22cf266028f676f98b5670e1193e33d571eb54f4c9f99f991a2d72",pid="1210248",source="bpf",vm_id=""} 73265
kepler_process_cpu_instructions_total{command="postgres: background writer",container_id="fbe5d970ec22cf266028f676f98b5670e1193e33d571eb54f4c9f99f991a2d72",pid="1210245",source="bpf",vm_id=""} 21080
kepler_process_cpu_instructions_total{command="postgres: walwriter",container_id="fbe5d970ec22cf266028f676f98b5670e1193e33d571eb54f4c9f99f991a2d72",pid="1210247",source="bpf",vm_id=""} 20648
```

We actually have access to the full set of commandline args, but this might be too verbose for use as labels, unless we can find a smarter way of adding them without increasing cardinality.

Re: Performance. It is expensive yes and would benefit from a cache since these values never change between collections.

Reply:
Cache added.